### PR TITLE
search: move logging from evaluateLeaf to Results

### PR DESCRIPTION
**Pull request by stefanhengl**
This moves logging to Prometheus, Honeycomb, and the Ping DB from
`evaluateLeaf` up to `Results`.

Apart from just moving around code I also changed how we keep track of
the elapsed time. I replaced `SearchResultsResolver.start` with
`SearchResultsResolver.elapsed`. The data shouldn't change too much, if
at all, with this change.

I also split up the switch block so that it becomes (hopefully) more
obvious when we can expect `srr` not to be `nil`.

### CODE REVIEW

**Review by rvantonder — COMMENTED** [review]
I'll have a deeper look tomorrow but I took a sneak peak at the draft and it looks good, I think the things are moved to the right place here.

File: cmd/frontend/graphqlbackend/search_results.go [hunk 1]
```diff
@@ -1463,24 +1475,24 @@ func (r *searchResolver) determineRepos(ctx context.Context, tr *trace.Trace, st
...
 	tr.LazyPrintf("searching %d repos, %d missing", len(resolved.RepoRevs), len(resolved.MissingRepoRevs))
 	if len(resolved.RepoRevs) == 0 {
 		alert := r.alertForNoResolvedRepos(ctx)
-		return searchrepos.Resolved{}, &SearchResultsResolver{db: r.db, alert: alert, start: start}, nil
+		return searchrepos.Resolved{}, &SearchResultsResolver{db: r.db, alert: alert}, nil
 	}
 	if resolved.OverLimit {
 		alert := r.alertForOverRepoLimit(ctx)
-		return searchrepos.Resolved{}, &SearchResultsResolver{db: r.db, alert: alert, start: start}, nil
+		return searchrepos.Resolved{}, &SearchResultsResolver{db: r.db, alert: alert}, nil
```
Comment (keegancsmith) [inline cmd/frontend/graphqlbackend/search_results.go:1495]: what do we log in these cases? I suppose we never logged them before? Would be useful to still know alerts in our logs/etc :) But given we likely never logged these cases, we should probably do in another PR?

**Review by keegancsmith — APPROVED** [review]

File: cmd/frontend/graphqlbackend/search_results.go (hunk 1 above)
Comment (stefanhengl) [reply cmd/frontend/graphqlbackend/search_results.go:1495]: I don't think we logged anything before, but now we do. The alert will be logged to Prometheus and Honeycomb, see lines 854 ff.
